src/OpenTelemetry/Trace/TracerProviderBuilderExtensions.cs

```diff
@@ -71,41 +71,31 @@ public static TracerProviderBuilder AddProcessor(this TracerProviderBuilder trac
             return tracerProviderBuilder;
         }
 
-        public static TracerProvider Build(this TracerProviderBuilder tracerProviderBuilder)
-        {
-            if (tracerProviderBuilder is TracerProviderBuilderSdk tracerProviderBuilderSdk)
-            {
-                return tracerProviderBuilderSdk.Build();
-            }
-
-            return null;
-        }
-
         /// <summary>
-        /// Adds a DiagnosticSource based instrumentation.
-        /// This is required for libraries which is already instrumented with
-        /// DiagnosticSource and Activity, without using ActivitySource.
+        /// Adds an activity listener for activities created the legacy way with the provided OperationName. The listener samples and processes these activities
+        /// using the Sampler and Processors provided to the TracerProvider.
         /// </summary>
-        /// <typeparam name="TInstrumentation">Type of instrumentation class.</typeparam>
         /// <param name="tracerProviderBuilder">TracerProviderBuilder instance.</param>
-        /// <param name="instrumentationFactory">Function that builds instrumentation.</param>
+        /// <param name="operationName">OperationName to add.</param>
         /// <returns>Returns <see cref="TracerProviderBuilder"/> for chaining.</returns>
-        internal static TracerProviderBuilder AddDiagnosticSourceInstrumentation<TInstrumentation>(
-            this TracerProviderBuilder tracerProviderBuilder,
-            Func<ActivitySourceAdapter, TInstrumentation> instrumentationFactory)
-            where TInstrumentation : class
+        public static TracerProviderBuilder AddLegacyActivityOperationName(this TracerProviderBuilder tracerProviderBuilder, string operationName)
```

Comment:
```suggestion
        public static TracerProviderBuilder AddLegacyActivity(this TracerProviderBuilder tracerProviderBuilder, string operationName)
```

Reply:
I have updated the name to AddLegacyActivity and updated the public API docs as well.